From 5f7c87d67de8623984b1cb35bb54639bc1bc4cf4 Mon Sep 17 00:00:00 2001
From: Song Zhao <song.zhao@amlogic.com>
Date: Mon, 4 May 2020 11:54:27 -0700
Subject: [PATCH] g12a/tm2: adjust memory layout

Set memory range for security
Increase ion/cma size for RDK4
---
 .../arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts | 5 +++--
 .../arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts | 6 +++---
 .../arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts | 6 +++---
 3 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
index 94ddce32e55a..a35429c40917 100644
--- a/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/g12a_s905x2_u212_drm_buildroot.dts
@@ -75,7 +75,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x8000000>;
+			size = <0x20000000>;
 			alignment = <0x400000>;
 		};
 		//di_reserved:linux,di {
@@ -112,8 +112,9 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
+			alloc-ranges = <0x0 0x30000000>;
 			linux,contiguous-region;
 		};
 		/* codec shared reserved */
diff --git a/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
index ce8e89785695..df8b6d643ce2 100644
--- a/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_t962e2_ab311_drm_buildroot.dts
@@ -82,10 +82,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x30000000 0x50000000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -99,7 +99,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x2000000>;
+			size = <0x20000000>;
 			alignment = <0x400000>;
 		};
 
diff --git a/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts b/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
index 0c16793f0a78..64eca2beb5e2 100644
--- a/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
+++ b/arch/arm/boot/dts/amlogic/tm2_t962x3_ab301_drm_buildroot.dts
@@ -91,10 +91,10 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			/* ion_codec_mm max can alloc size 80M*/
-			size = <0x13400000>;
+			size = <0x18400000>;
 			alignment = <0x400000>;
 			linux,contiguous-region;
-			alloc-ranges = <0x30000000 0x50000000>;
+			alloc-ranges = <0x0 0x30000000>;
 		};
 
 		/* codec shared reserved */
@@ -108,7 +108,7 @@
 		ion_cma_reserved:linux,ion-dev {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0x2000000>;
+			size = <0x20000000>;
 			alignment = <0x400000>;
 		};
 
-- 
2.24.1

